.pre-commit-config.yaml

```diff
@@ -50,8 +52,16 @@ repos:
         language: system
         entry: cargo xtask gen-man
         stages: [commit, merge-commit, manual]
+        pass_filenames: false
       - id: gen-comp
         name: gen-comp
         language: system
         entry: cargo xtask gen-comp
         stages: [commit, merge-commit, manual]
+        pass_filenames: false
+      - id: gen-openapi
+        name: gen-openapi
+        language: system
+        entry: cargo xtask gen-openapi
+        stages: [commit, merge-commit, manual]
+        pass_filenames: false
```

Comment:
Ideally this would be a vite plugin rather than a pre-commit hook, but... vite's plugin API is honestly quite bewildering to me at the moment.

Reply:
Agreed. This is part of the build process and not necessarily a pre-commit "check".

Reply:
There's also a bit of a circular dependency problem.. to avoid that we'd need to split `cockpit-api` into a separate crate that knows how to emit its API, which we then combine with the `cockpit-ui` to build the `cockpitd` server binary that people run.

Reply:
Happy to look at this at some point, but I think it deserves its own tech-debt ticket rather than blocking this one.

Reply:
Yes, would be great if you can look into this. But I agree, this shouldn't be a blocker for this PR.